products/platform/documentation-site/research/search-design-doc.md

```diff
@@ -1,15 +1,156 @@
+
+### Caveats
+
+- Access Control: By making the documentation landing page public, it becomes difficult, if not impossible, to allow an unauthenticated user to search documentation from private sources (like a private Github repo). One suggestion is that search results lead to public documentation sources, and we include links to private documentation in the public sources. That way, we can rely on existing access control solutions, while still providing a decent solution. One of the major downsides of that approach is auditing, adding, updating, and maintaining links to/from the public and private sources.
+
+### Security Concerns
+
+If the search input dispatched a request directly to the Algolia API, then there'd be minimal risk of the documentation search negatively affecting other systems. If the request was proxied through our API, then we should be able to rely on any existing DOS protections.
+
+### Privacy Concerns
+
+While there shouldn't be any PII in our public documentation sources, it is possible it exists, and hasn't been discovered yet. Therefore, it is possible that our search results could lead public users to sources that contain PII. If PII is discovered in a repo, we have a [checklist](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/removing-sensitive-data-from-a-repository.md#checklist-for-removing-pii-in-md-file-from-a-documentation-repo) for how to remove it.
+
+### Open Questions and Risks
+
+- Since the documentation landing page will be public only, we will not be able to search private sources of documentation. If searching private documentation becomes a necessary feature in the future, the proposed implementation will likely not be sufficient.
+- How often should we reindex our documentation sources? For perspective, Algolia's open source crawler runs every 24 hours.
+- What should the URL of the documentation landing page be? Since we have `design.va.gov`, does something like `docs.va.gov` make sense?
+- What is the current opportunity cost of building a custom solution? What are we choosing to delay right now by building a custom solution?
+- What is the future opportunity cost of maintaining a custom solution? What will we have to delay in the future by maintaining a custom solution?
+- What is the key person risk? How disruptive would it be if the author(s) of the custom solution were temporarily/permanently unavailable?
+- What is the Not-Invented-Here risk? How likely is that we're reinventing the wheel?
+- What is the inexperience risk? Have any of the team members involved ever built/deployed/maintained this type of system before?
+- It's possible that work estimates are materially inaccurate.
+- What is the documentation debt risk? At least one person interviewed during the discovery sprint said that we need a solution ASAP. Is there a risk that people are deferring documentation in the present until the custom documentation solution is shipped?
+
```

Comment:
- How do we configure what pages get indexed? 
- What is a page? What parts of the page get indexed? 
- How the data on the pages gets transformed into algolia's data structure e.g. how do we configure the titles on the results 
- How are results weighted? How do we control the weights / priorities of results? 

![Screen Shot 2020-02-03 at 14 10 46](https://user-images.githubusercontent.com/4998130/73682773-0f594e00-468f-11ea-940b-403f469d5c23.png)

would like to see a plan that generates the index and demonstrates the results locally- before we commit to building infrastructure to support this. At the very least we should user test this with VSP people to prove that this index is more useful than the one provided by github.

Reply:
Is there any sort of data munging that has to occur in our set up or do we expect to try to control the index / results solely from the source data?

Reply:
Here are some links to the crawler config documentation. 
https://community.algolia.com/docsearch/required-configuration.html
https://community.algolia.com/docsearch/config-file.html

To explicitly tell the crawler which routes to start crawling from, we can add the routes to the the [`start_urls` array](https://community.algolia.com/docsearch/config-file.html#start_urls).

To prevent the crawler from crawling a specific route, we can add the route to the [`stop_urls` array](https://community.algolia.com/docsearch/config-file.html#stop_urls-optional). 

To exclude specific parts of a page from the crawler, we can add CSS selectors to the [`selectors_exclude` array](https://community.algolia.com/docsearch/config-file.html#selectors_exclude-optional)

You can read a detailed explanation about how Algolia extracts payloads from selectors [here](https://community.algolia.com/docsearch/how-do-we-build-an-index.html). 

According to https://community.algolia.com/docsearch/required-configuration.html#use-the-right-classes-as-selectors:
> `lvlX` selectors should use the standard title tags like `h1`, `h2`, `h3`, etc. You can also use static class. Set a unique `id` or `name` attribute to these elements as detailed below.
> 
> Every DOM elements matching the `lvlX` selectors must have a unique `id` or `name` attribute. This will help the redirection to directly scroll down to the exact place of the matching elements. These attributes defined the right anchor to use.
> 
> Every textual element (selector text) must be wrapped in a tag `<p>` or `<li>`. This content must be atomic and split into small entities. Be careful to never nest one matching elements into another one, it will create duplicates.

GitHub's markdown seems to properly render the heading and textual elements that the Algolia scraper looks for.
![image](https://user-images.githubusercontent.com/6130520/73795721-f4f7a100-4770-11ea-99fa-1597dfa0ff3c.png)